restaurant/services.py

```diff
@@ -0,0 +1,71 @@
+from datetime import timedelta
+
+from django.urls import reverse
+
+from .tasks import send_reservation_email
+
+
+def build_reservation_mail(reservation, action, request=None):
+    relative_url = reverse("restaurant:reservation_update", args=[reservation.pk])
+    url = request.build_absolute_uri(relative_url) if request else relative_url
+    name = reservation.owner.get_full_name()
+
+    if action == "created":
+        subject = "НОВЫЙ РЕЗЕРВ!!!"
+        body = (
+            f"НОВЫЙ РЕЗЕРВ!\n\n"
+            f"Дата: {reservation.date}\n"
+            f"Время: {reservation.time}\n"
+            f"Гость: {name} - {reservation.owner}\n"
+            f"Столик №{reservation.table}\n"
+            f"Комментарий: {reservation.comment}\n"
+            f"Ссылка для подтверждения: {url}"
+        )
+
+    elif action == "updated":
+        subject = "ИЗМЕНЕНИЕ РЕЗЕРВА!!!"
+        body = (
+            f"ИЗМЕНЕНИЯ РЕЗЕРВА!\n\n"
+            f"Дата: {reservation.date}\n"
+            f"Время: {reservation.time}\n"
+            f"Гость: {name} - {reservation.owner}\n"
+            f"Столик №{reservation.table}\n"
+            f"Комментарий: {reservation.comment}\n"
+            f"Ссылка для подтверждения: {url}"
+        )
+
+    elif action == "deleted":
+        subject = "РЕЗЕРВ БЫЛ ОТМЕНЁН!!!"
+        body = (
+            f"ОТМЕНА РЕЗЕРВА!\n\n"
+            f"Дата: {reservation.date}\n"
+            f"Время: {reservation.time}\n"
+            f"Гость: {name} - {reservation.owner}\n"
+            f"Столик №{reservation.table}\n"
+            f"Комментарий: {reservation.comment}\n"
+        )
+
+    else:
+        subject = body = None
+
+    return subject, body
+
+
+def send_reservation_notification(reservation, action, user=None, request=None):
+    if user and user.pk != reservation.owner.pk:
```

Comment:
Так можно сэкономить один запрос в базу
```suggestion
    if user and user.id != reservation.owner_id:
```